app/controllers/admin/access_controls_controller.rb

```diff
@@ -77,4 +97,29 @@ def assign
   private def set_access_control
     @access_control = access_control_scope.find(params[:id].to_i)
   end
+
+  private def access_controls
+    @access_controls ||= AccessControl.all
+  end
+
+  private def histories
+    @histories ||= [].tap do |histories|
+      access_controls.each do |access_control|
+        history = Audit::Versions.new(access_control, access_control_component_config)
+        histories << history
+      end
+    end
+  end
+
+  private def data
+    @data ||= histories.map do |history|
+      versions = history.version_scope
+      history.wrap_display_versions(versions).map do |version|
+        {
+          history: history,
+          version: version,
+        }
+      end
+    end.flatten(1)
```

Comment:
Please double check, but I think this is just:

```suggestion
    @data ||= histories.flat_map do |history|
      versions = history.version_scope
      history.wrap_display_versions(versions).map do |version|
        {
          history: history,
          version: version,
        }
      end
    end
```